utilities/pipelines/resourceRemoval/helper/Invoke-ResourceRemoval.ps1

```diff
@@ -24,12 +27,21 @@ function Invoke-ResourceRemoval {
         [string] $ResourceId,
 
         [Parameter(Mandatory = $true)]
-        [string] $Type
+        [string] $Type,
+
+        [Parameter(Mandatory = $false)]
+        [array] $Locks = @()
```

Comment:
Isn't it enough to add the lock resource type as a first element of the order of resources to be deleted in Initialize-DeploymentRemoval? (right above the diagnostic)

Reply:
True story. Applied & tested successfully.